oneargopy/Argo.py

```diff
@@ -183,98 +419,494 @@ def __try_download(self, file_name: str, update_status: bool)-> None:
+            # end_date is optional and should be set to tomorrow if not provided
+            if self.end_date != None:
+                self.end_date = datetime.fromisoformat(self.end_date).replace(tzinfo=timezone.utc)
+            else:
+                self.end_date = datetime.now(timezone.utc) + timedelta(days=1)
+        except ValueError:
+            print(f" Start date: {self.start_date} or end date: {self.end_date} is not in the expected format 'yyyy-mm-dd'")
+        
+        # Validate datetimes
+        if self.start_date > self.end_date:
+            if self.download_settings.verbose: 
+                print(f'Current start date: {self.start_date}')
+                print(f'Current end date: {self.end_date}')
+            raise ValueError(f'The start date must be in the past when compared to the end date.')
+        if self.start_date < datetime(1995, 1, 1, tzinfo=timezone.utc):
+            if self.download_settings.verbose: print(f'Current start date: {self.start_date}')
+            raise ValueError(f'Start date must be after at least: {datetime(1995, 1, 1, tzinfo=timezone.utc)} for any floats to be active.')
+        
+        # Set to datetime64 for dataframe comparisons
+        self.start_date = np.datetime64(self.start_date)
+        self.end_date = np.datetime64(self.end_date)
+
+
+    def __validate_outside_kwarg(self): 
+        """ A function to validate the value of the 
+            optional 'outside' keyword argument.
+        """
+        if self.download_settings.verbose: print(f"Validating 'outside' keyword argument...")
+
+        if self.outside is not None: 
+            if self.outside != 'time' and self.outside != 'space' and self.outside != 'both':
+                raise Exception(f"The only acceptable values for the 'outside' keyword argument are 'time', 'space', and 'both'.")
+            
+    
+    def __validate_type_kwarg(self): 
+        """ A function to validate the value of the 
+            optional 'type' keyword argument.
+        """
+        if self.download_settings.verbose: print(f"Validating 'type' keyword argument...")
+
+        if self.float_type != 'all' and self.float_type != 'phys' and self.float_type != 'bgc':
+                raise Exception(f"The only acceptable values for the 'type' keyword argument are 'all', 'phys', and 'bgc'.")
+        
+    
+    def __validate_floats_kwarg(self):
+        """ A function to validate the 'floats' keyword argument. 
+            The 'floats' must be a list even if it is a single float.
+        """
+        # Casting to list
+        if not isinstance(self.floats, list):
+            self.floats = [int(self.floats)]
+
+        # Checking that floats are in index dataframes
+        all_present = all(float in self.prof_index['wmoid'].values for float in self.floats)
+        if not all_present: 
+            raise Exception(f"You have passed float values that do not exist in the dataframes: {self.floats}")
```

Comment:
Can you show only the float IDs that are not present in the index? (easy test: pass in 1,4903500; 1 should be listed)

Reply:
![image](https://github.com/lovelysandlonelys/OneArgo-Python-Dev/assets/91633941/21892d69-3afb-460b-adb9-98f7f95df36c)

![image](https://github.com/lovelysandlonelys/OneArgo-Python-Dev/assets/91633941/af2a47c4-c7b7-45b6-8fb2-c6151b2662dd)